### ISSUE TITLE
Failures on Allow empty switch expressions.

### ISSUE THREAD

**Issue by stereotype441** [dart-lang/sdk#51847]
There are new test failures on [Allow empty switch expressions.](https://dart-review.googlesource.com/c/sdk/+/290907).

The tests
```
language/patterns/empty_switch_expression_weak_test Crash (expected Pass)
```
are failing on configurations
```
dart2js-hostasserts-linux-d8-unsound
```

**Comment by rakudrama** [comment]
Fixed via 6bdf3c042a473f6df5da7b05a1beb5e5eb249996

**Comment by rakudrama** [comment]
The fix had to be reverted.
